test/python/transpiler/test_calibrationbuilder.py

```diff
@@ -57,15 +57,21 @@ def setUp(self):
         def _get_cr(time_inst, name):
             return isinstance(time_inst[1], Play) and time_inst[1].pulse.name.startswith(name)
 
-        cx_sched = self.inst_map.get("cx", (0, 1))
+        qubits = (0, 1)
```

Comment:
Why this must be changed? This instmap is the past snapshot of Hanoi which must be the forward CX. You need to upgrade this test class with `ddt` to run both on Hanoi and Sherbrooke to test CX and ECR.

Reply:
I originally changed `cx_sched` to `cr_sched` to treat the native entangling gates on equal footing. But I think you are correct in that I would need to make a separate test class with `ddt` to test the `ecr` case with (fake) Sherbrooke. Might it make more sense to run the same code for the two cases of backends (also this assumes Hanoi will always default to `cx` and Sherbrooke to `ecr`)?

Reply:
ddt with two backends sounds better, but you need overhaul of the test class because it is currently tied to Hanoi. You can remove setup and add a method to get individual play instruction, e.g. `self.u0p_play` -> `def get_u0p_play(self, backend):` and then pass the backend with ddt at each test function.

Hanoi and Sherbrooke might change in future, but [fake backend data](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/providers/fake_provider/backends/hanoi/defs_hanoi.json) is stored as a part of terra and the object is built from there. So the gate will be always cx for Hanoi (and I guess ecr for Sherbrooke) unless someone manually update the backend data (there is a command to update the data).